src/metabase/sync/sync_metadata/tables.clj

```diff
@@ -91,10 +98,20 @@
               {:details
                (assoc (:details database) :version (:version db-metadata))}))
 
-(defn- cruft-dependent-columns [table-name]
-  ;; if this is a crufty table, mark initial sync as complete since we'll skip the subsequent sync steps
-  (let [is-crufty? (is-crufty-table? table-name)]
-    {:initial_sync_status (if is-crufty? "complete" "incomplete")
+(mu/defn- cruft-dependent-cols [{table-name :name :as table}
+                                database
+                                sync-stage :- [:enum ::reactivate ::create ::update]]
+  ;; if this is a crufty table, mark initial sync as complete since we'll be skipping the subsequent sync steps
+  (let [is-crufty? (crufty/name? table-name {:patterns        crufty-table-patterns
+                                             :pattern-strings (some-> database :settings :auto-cruft-tables)})]
+    {:initial_sync_status (cond
+                            ;; if a table exists, we don't want to overwrite the initial sync status
+                            ;; while updating its table metadata, so that it is "complete" during the sync.
+                            ;; See: [[metabase.sync.util-test/initial-sync-status-table-only-test]]
+                            (= sync-stage ::update) (:initial_sync_status table)
+                            ;; if a table is crufty, we mark it as complete to skip the subsequent sync steps
+                            is-crufty?              "complete"
```

Comment:
does this actually work? if so I'm slightly surprised, I thought the sync status was just to power the UI

Reply:
Hmm, I believed this comment, but maybe it's no longer true (or never was). I dug around to see where the comment came from originally. [it is from this commit](https://github.com/metabase/metabase/blob/ffbb526e5deabbb668cc62cf363af2f4ecaac720/src/metabase/sync/sync_metadata/tables.clj#L99). I don't see anywhere that we are filtering for a table's initial sync status in the sync steps. 

So: I think it does get synced, but just marked crufty (hidden) which is fine for this since we want to allow admins to unhide tables that get marked crufty incidentally.

Reply:
I'll cleanup the comment